### ISSUE TITLE
Being able to set account data through /commands is confusing

### ISSUE THREAD

**Issue by jplatte** [vector-im/element-web#16241]
### Description

I just wanted to change my IRC nick and despite being a Matrix user for years and an IRC user before that for quite a while too, I made two mistakes before getting it to work:

* Tried setting the nick through the NickServ chat rather than contacting the appservice bot (not Element's fault in any way)
* Tried setting the nick with `/nick` instead of `!nick` (what this report is about)

### Steps to reproduce

- Type `/nick SomeNick` in any room
- Observe your global display name changing, generating visible state events in every joined room

I would have expected an error, or at least a confirmation prompt. Changing the display name feels too disruptive to be so easy to do. If there was a prompt, it could also offer the option to change the display name for just the current room, making room-specific display names a tad easier to discover (if you deem that useful).

### Version information

Seems irrelevant but

- **Platform**: Desktop
- **OS**: arch linux
- **Version**: 1.7.16

**Comment by t3chguy** [comment]
Having prompts on slash commands is a bad smell IMO, slash commands are power user features and are very intentional, confirming them should be reserved to only destructive ones

**Comment by jplatte** [comment]
Do you disagree that this is an issue? I don't really care *how*, but IMHO this should to be fixed somehow (otherwise I wouldn't have created this issue report, of course).

**Comment by t3chguy** [comment]
Its akin to expecting Linux commands like `rm` always asking for your confirmation, its an explicit action with a purpose, it doesn't need an additional prompt in my opinion

**Comment by jplatte** [comment]
> Its akin to expecting Linux commands like `rm` always asking for your confirmation

It is. And I'm pretty sure the majority of people who semi-regularily use `rm` (without having it aliased to `rm -i` / `rm -I`) has deleted something they shouldn't have before. It's not at all the kind of tool I would expect in a modern software that has a `ui/ux` GitHub issue label.

**Comment by t3chguy** [comment]
Actually poor example as I said

> confirming them should be reserved to only destructive ones

Its more akin to `mv` which doesn't even have an optional prompt except in the case of conflicts.

**Comment by jplatte** [comment]
Well yeah, `/nick` is not destructive (unless it also overrides room display names, in which case this would be even more problematic), but doing it accidentally is still very much annoying.

Anyways, it doesn't seem like we'll find an agreement among the two of us. I am curious what others think.

**Comment by t3chguy** [comment]
Indeed, for people with opinions, best to leave up/down votes on the top of this issue to convey them

**Comment by HarHarLinks** [comment]
> (unless it also overrides room display names, in which case this would be even more problematic)

afaik, exactly this is the case, I had to deal with this myself yesterday. display names including room display names need a complete revamp (or rather design at all?), but that is not exactly the issue here so let's disregard this for the time being:

slash commands fall into several categories, including but not limited to: bad/missing other UI (`/rainbow`, `/confetti`), IRC/power user (`/invite` and other room management), debug stuff (`/devtools`), misc (`/ddg` which is only not a command, but also an ad?). In addition there is account data (`/nick`, `/myavatar`) which you (could) argue falls in the IRC/power user category, but all other commands in that category apply to the current room with exception of those that explicitly take a room id as argument.

thus I agree with jplatte that it is too easy to do accidentally and should get a warning popup (checkbox to never show again?) or be renamed `/myglobalnick`. Maybe commands should be reworked a bit regarding such categories 🤷

**Comment by t3chguy** [comment]
> (unless it also overrides room display names, in which case this would be even more problematic)

That's Synapse behaviour, not Element. Element just updates your default displayname and Synapse feels appropriate to copy that to all rooms, including those you have a different name in.

**Comment by HarHarLinks** [comment]
In fact the spec feels that is appropriate and required, and I did bring this up on matrix-doc recently. The `/nick` command triggers exactly that at this point and thus should be assessed in relation to current spec, no? It is not even possible to work around that from just Element side.

**Comment by jryans** [comment]
> Observe your global display name changing, generating visible state events in every joined room

Personally, I'd suggest instead fixing this symptom so that it doesn't generate visible events, via changes like https://github.com/vector-im/element-web/issues/12063 and similar.

**Comment by jplatte** [comment]
Personally, I would find this annoying even without it adding an unread marker for others.
